File: Cpp/1481_Least_Number_of_Unique_Integers_after_K_Removals.cpp
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int,int> m;
        multiset<int> s;
        for(auto& num : arr) {
            m[num] += 1;
        }
        
        for(auto& it : m) {
            s.insert(it.second);
        }
        
        int result = s.size(); 
        int i = k;
        for(auto& it : s) {
            if(it > i) break;
            else i -= it;
            --result;
        }
        
        return result;
    }
};

File: Kotlin/1481_Least_Number_of_Unique_Integers_after_K_Removals.kt
class Solution {
    fun findLeastNumOfUniqueInts(arr: IntArray, k: Int): Int {
        val m = arr.asList().groupingBy { it }.eachCount()
        val sorted = m.values.sorted()
        
        var result = sorted.size
        var i = k
        for(count in sorted) {
            if(count > i) break
            else i -= count
            --result
        }
        
        return result
    }
}